modules/Core/src/main/java/org/terasology/core/debug/AnimationScreen.java

```diff
@@ -0,0 +1,127 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.characters.StandComponent;
+import org.terasology.logic.characters.WalkComponent;
+import org.terasology.logic.common.InspectionToolComponent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.widgets.*;
+
+import java.util.ArrayList;
+
+/**
+ * A Tool for testing the animations of entities.
+ * Animation speed can be edited with this utility
+ */
+public class AnimationScreen extends CoreScreenLayer {
+
+    private static final Logger logger = LoggerFactory.getLogger(AnimationScreen.class);
+    @In
+    private EntityRef entityRef;
+    @In
+    private LocalPlayer localPlayer;
+    @In
+    private EntityManager entityManager;
+    @In
+    private AssetManager assetManager;
+    private UIButton spawnEntityIdButton;
+    private UISlider animationSpeedSlider;
+    private UIDropdownScrollable<ResourceUrn> entityDropdown;
+
+    @Override
+    public void initialise() {
+        spawnEntityIdButton = find("spawnEntityIdButton", UIButton.class);
+        entityDropdown = find("entityDropdown", UIDropdownScrollable.class);
+        logger.info("Number of available skeletal meshes: " + assetManager.getAvailableAssets(SkeletalMesh.class).size());
+        //ArrayList skeletalMesh = );
+        if (entityDropdown != null ) {
+            entityDropdown.setOptions(new ArrayList(assetManager.getAvailableAssets(SkeletalMesh.class)));
+        }
+        animationSpeedSlider = find("entityAnimationSpeedSlider", UISlider.class);
+        if (animationSpeedSlider != null) {
+            animationSpeedSlider.setMinimum(-0.0f);
+            animationSpeedSlider.setIncrement(0.1f);
+            animationSpeedSlider.setRange(10.0f);
+            animationSpeedSlider.setPrecision(1);
+        }
+        spawnEntityIdButton.subscribe(widget -> {
+            Vector3f localPlayerPosition = localPlayer.getPosition();
+            Quat4f localPlayerRotation = localPlayer.getRotation();
+            Optional<Prefab> prefab = assetManager.getAsset(entityDropdown.getSelection(), Prefab.class);
+            if (prefab.isPresent() && prefab.get().getComponent(LocationComponent.class) != null) {
```

Comment:
You may want to add an `else` leg here and/or put the remainder of the live code inside the if's body - in the else case `entityRef` wouldn't be initialized and likely the later lines would cause a crash.